From 591a8b84031204265221672d61b8fb5c80bda21e Mon Sep 17 00:00:00 2001
Message-ID: <591a8b84031204265221672d61b8fb5c80bda21e.1711915492.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1711915492.git.sharlatanus@gmail.com>
References: <cover.1711915492.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 31 Mar 2024 13:15:14 +0100
Subject: [PATCH 04/12] gnu: Add go-code-cloudfoundry-org-bytefmt.

* gnu/packages/golang-xyz.scm (go-code-cloudfoundry-org-bytefmt): New
variable.

Change-Id: I1d77dd49987ad09b15149f244024831d26780e6e
---
 gnu/packages/golang-xyz.scm | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 1bf296bda0..415233b100 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -68,6 +68,37 @@ (define-module (gnu packages golang-xyz)
 ;;; Libraries:
 ;;;
 
+(define-public go-code-cloudfoundry-org-bytefmt
+  (package
+    (name "go-code-cloudfoundry-org-bytefmt")
+    (version "0.0.0-20240329144308-0c372429d24b")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/cloudfoundry/bytefmt")
+             (commit (go-version->git-ref version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0aqzbiy3idddyj39i7ydkjhnmpcbwr99g094kqiw72m9flrvrnxj"))
+       (snippet
+        #~(begin (use-modules (guix build utils))
+                 (delete-file-recursively "vendor")))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:go go-1.20
+      #:import-path "code.cloudfoundry.org/bytefmt"))
+    (native-inputs
+     (list go-github-com-onsi-gomega
+           go-github-com-onsi-ginkgo-v2))
+    (home-page "https://pkg.go.dev/code.cloudfoundry.org/bytefmt")
+    (synopsis "Human readable byte formatter for Golang")
+    (description
+     "Package bytefmt contains helper methods and constants for converting to and from
+a human-readable byte format.")
+    (license license:asl2.0)))
+
 (define-public go-github-com-a8m-envsubst
   (package
     (name "go-github-com-a8m-envsubst")
-- 
2.41.0

